Jetpack Manage: Implement submit product feedback

yashwin opened the pull request:
Resolves https://github.com/Automattic/jetpack-genesis/issues/165
Resolves https://github.com/Automattic/jetpack-genesis/issues/166

## Proposed Changes

This PR implements submit product feedback.

## Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb.

**Instructions**

1. Apply patch: D133680-code
2. Visit the Jetpack Cloud link > Append the URL with `?flags=jetpack/user-feedback-form` 
3. Click the 'Share user feedback' menu item on the bottom left of the sidebar.

<img width="292" alt="screen-shot-2023-12-20-at-7 39 21-pm" src="https://github.com/Automattic/wp-calypso/assets/56598660/eef6bf7f-7106-43aa-ad02-0d83b3340313">

4. Enter the feedback text and select the rating.
5. Click the `Submit your feedback` button > Verify that the busy state of the button is displayed when the request is being made > Verify that the feedback is submitted with the fields: `rating`, `feedback`, and `source_url`. `source_url` is the current page the user is viewing. 

<img width="867" alt="Screenshot 2024-01-04 at 3 46 20 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/8e8c3cf5-a61c-41c3-a2ec-72efd04bc63d">


6. Also, verify that the success message is displayed once the feedback is submitted and you have received an email for the same.

<img width="372" alt="Screenshot 2024-01-05 at 9 08 07 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/628e56a2-8012-412f-a3f1-fa880b0e3782">

<img width="1289" alt="Screenshot 2024-01-05 at 9 18 12 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/07bd154f-2de7-4b90-9c93-bdfb866e13aa">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

yashwin committed:
Implement submit product feedback

jkguidaven commented on client/jetpack-cloud/components/user-feedback-modal-form/index.tsx:
```diff
@@ -37,9 +40,10 @@ export default function UserFeedbackModalForm( { show, onClose }: Props ) {
-		// TODO: send feedback to backend
-	}, [ hasCompletedForm ] );
+		// Remove the hash from the URL.
+		const sourceUrl = window.location.href.split( '#' )[ 0 ];
```
I suggest combining `window.location.origin` and `window.location.pathname` to strip down any query strings after **?**. wdyt?

yashwin replied:
Makes sense. I have now updated it.

yashwin committed:
Close modal after a successful request

yashwin commented:
Thanks for your review, @jkguidaven! I have addressed your comment. 

Also, this PR is now ready for a full review. I have also made some new changes: https://github.com/Automattic/wp-calypso/pull/85975/commits/dd8d625cbc3f11a3ef193432a20a6ef020123332

jkguidaven commented on client/jetpack-cloud/components/user-feedback-modal-form/index.tsx:
```diff
@@ -19,10 +22,32 @@ const DEFAULT_RATING_VALUE = 0;
+					duration: 5000,
+				} )
+			);
+			onModalClose?.();
```
[non-blocker] `onModalClose` here is always defined, so we can omit `?.()`.

yashwin replied:
🤦

yashwin replied:
Fixed: https://github.com/Automattic/wp-calypso/compare/dd8d625cbc3f11a3ef193432a20a6ef020123332..c6ac6b03c79e45dc211657e4eb70a20af30ab973

jkguidaven commented on client/jetpack-cloud/components/user-feedback-modal-form/index.tsx:
```diff
@@ -100,6 +119,7 @@ export default function UserFeedbackModalForm( { show, onClose }: Props ) {
 
 			<div className="user-feedback-modal-form__footer">
 				<Button
+					busy={ isSubmittingFeedback }
```
👍

jkguidaven reviewed (approved):
Thanks for addressing my comment. I tested it and I can confirm the request is submitted. I left 1 non-blocking comment but this looks good to me 👍
